### PR TITLE
Added health management, and HP bars for characters.

### PR DESCRIPTION
**Pull request by milosmilosevic13**
Health is now a separate component. Health management system takes care of displaying HP bars under characters as well as removing characters whose HP drops below zero.

### CODE REVIEW

File: source/dagger/gameplay/team_game/hp_system.cpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
+#include "hp_system.h"
+
+#include "core/engine.h"
+#include "core/graphics/sprite.h"
+#include "core/game/transforms.h"
+
+#include <iostream>
```
Comment (mmacura9) [inline source/dagger/gameplay/team_game/hp_system.cpp:7]: I think you do not use this.

File: source/dagger/gameplay/team_game/mage.cpp [hunk 2]
```diff
@@ -103,7 +105,7 @@ void ancient_defenders::MageBehaviorSystem::Run()
             else {
                 AnimatorPlay(animation_, "ancient_defenders:mage:ATTACK_FRONT");
             }
-            Engine::Registry().get<Enemy>(range_.target).health -= mage_.meleeDmg;
+            Engine::Registry().get<Health>(range_.target).currentHealth -= mage_.meleeDmg * Engine::DeltaTime();
```
Comment (mmacura9) [inline source/dagger/gameplay/team_game/mage.cpp:109]: This should be also added to golem.h/ golem.cpp

File: source/dagger/gameplay/team_game/hp_system.cpp (hunk 1 above)
Comment (milosmilosevic13) [reply source/dagger/gameplay/team_game/hp_system.cpp:7]: You are right, it's a leftover from some testing, will be removed.

File: source/dagger/gameplay/team_game/mage.cpp (hunk 2 above)
Comment (mmacura9) [reply source/dagger/gameplay/team_game/mage.cpp:109]: I mean Health should be added to golem

**Review by mmacura9 — APPROVED** [review]
Everything is fine, we just should add this to golem.

**Pull request merged**